src/h3lib/include/polygonAlgos.h

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file polygon.c
+ * @brief Polygon algorithms
+ */
+
+#include "polygon.h"
```

Comment:
If this is included multiple times, wouldn't we prefer to not have includes in it?

Reply:
I think this is safe, since the header files are wrapped in `#ifndef` statements. I can take the include out, probably not needed.

Reply:
I was thinking more about reading the preprocessed code, but I suppose it's true that that would be reduced away anyways.

Reply:
Yes the include guards help you for sure.

Reply:
This compiles fine without the `include` statements, but I'd rather leave them in if it doesn't have any significant impact, as they express the dependencies specific to this file. Otherwise it's up to the including file(s) to infer the dependencies and include them, which seems weird to me.